// rev-logger/server.js
/**
* Description: This is a service for BeaconURL
* Author: Rev Software
*/

/*********** For SysLog Integration **********/
var winston = require('winston');
require('winston-syslog2').Syslog;
var conf= require("/opt/revsw-cube/config/config");

//winston.add(winston.transports.Syslog, {colorize:true});

winston.setLevels(winston.config.syslog.levels);
winston.addColors(winston.config.syslog.colors );

//var transport = new winston.transports.Syslog();

var transport = new winston.transports.Syslog({
   // levels:winston.config.syslog.levels,
    level: conf.logging.syslog_level,
    id: 'hello syslog',
    facility: 'user',
    prettyPrint: true,
    colorize: true,
    showPid: true
});

var delogger = new (winston.Logger)({ transports: [
  new (winston.transports.File)({
    filename: conf.logging.debug_log_file_path,
    level:'debug',
    colorize: true ,
    facility: 'user',
    json:false,
    prettyPrint:true,
    showPid:true
    })
]});

//, new (winston.transports.Console)({ colorize: true });
/**var transportFile = new winston.transports.File({
    filename: '/var/log/somefile.log'
});*/

var logger = new winston.Logger({
  transports: [transport]
});
//logger.cli();
//logger.add({colorize: true}));

/*********** For SysLog Integration **********/

exports.log  = function(level,msg){	
  /**if(level=='debug') {
    if(conf.logging.is_debug_mode_on) {
      logger.log(level,msg);
    }
  } else {
      logger.log(level,msg);
  }
   logger.log(level,msg);
   */
    
   if(conf.logging.syslog_level=="debug" && level=="debug") {
    delogger.log("debug",msg);
   } else {
      logger.log(level,msg);
   }
};
